# Quiz 1

#### Name:


#### Grade:


#### Region:

Please answer the questions according to the following example.

### Example

What will be the output of the code below?

In [1]:
def addOne(x):
    return x + 1
    
print addOne(2)

3


#### Answer
`3`

### Problem 1
What will be the value of `x`?

In [2]:
x = 2*4+3*8
x

32

### Problem 2
What will be the value of `x`?

In [3]:
x = 2**5+7*3
x

53

### Problem 3
What will be the output of the code below?

In [4]:
x = 0
if True:
    x = 1
else:
    x = 2
print x

1


### Problem 4
What will be the output of the code below?

In [5]:
for x in [1, 2, 3]:
    print x

1
2
3


### Problem 5
What will be the output of the code below?

In [6]:
n = 5
while n != 0:
    print n
    n -= 1

5
4
3
2
1


### Problem 6
What will be the output of the code below?

In [7]:
n = 5
while n != 0:
    n -= 1
    print n

4
3
2
1
0


### Problem 7
What will be the output of the code below?

In [8]:
def mystery(x):
    return x*5
print mystery(10)

50


### Problem 8
What will be the output of the code below?

In [9]:
def mystery(x):
    return k*5
print mystery(10)

NameError: global name 'k' is not defined

### Problem 9
What will be the output of the code below?

In [10]:
n = 5
while n > 0:
    n -= 1
print n

0


### Problem 10
What is the value of `y`

In [11]:
def tripleIt(x):
    print x + x + x
y = tripleIt('123')

123123123


A. `'123'`

B. `'123123123'`

C. `369`

D. None of the above

### Problem 11
What will be the output of the code below?

In [12]:
# Problem 11
for n in [1, 2, 3, 4, 5, 6, 7]:
    if n % 3 == 0:
        print 'a'
    elif n % 4 == 0:
        print 'b'
    else:
        print 'c'

c
c
a
b
c
a
c


### Problem 12
What will be the output of the code below?

In [13]:
def doubleIt(x):
    return x + x
print Doubleit('123')

NameError: name 'Doubleit' is not defined

### Problem 13
What will be the output of the code below?

In [14]:
def doubleIt(x):
    return x + x + x + x
print doubleIt('123')

123123123123


### Problem 14
What will be the output of the code below?

In [15]:
x = [1, 2, 3, 4]
print x[1:3]

[2, 3]


### Problem 15
What will be the output of the code below?

In [16]:
x = [1, 2, 3, 4, 5, 6]
result = 0
for i in [0, 1, 2]:
    result += x[i]
print result

6


### Problem 16
What will be the output of the code below?

In [17]:
x = [7, 6, 5, 4, 3, 2, 1]
for i in x[0:len(x)/2]:
    print i

7
6
5


### Problem 17
What will be the output of the code below?

In [18]:
def mystery(x):
    return x + 2
    print x
print mystery(5)

7


### Problem 18

Circle the following which are of type boolean:

 A. `1` 
 
 B. `'1'`
 
 C. `'hello'`

 D. `x`

### **E. `True`**

 F. `true`

### ** G. `False`**

 H. `false`

 I. `'False'`

### Problem 19
What will be the output of the code below?

In [19]:
x = 54321
print x[3]

TypeError: 'int' object has no attribute '__getitem__'

### Problem 20
What will be the output of| the code below?

In [20]:
x = 'this is a string'
print len(x)

16


### Problem 21
What will be the output of the code below?

In [21]:
for i in range(2,12):
    print i
    if i % 3 == 0:
        break

2
3


### Problem 22
What will be the output of the code below?

In [22]:
for i in range(2,12):
    print i
    if i % 3 == 0:
        continue

2
3
4
5
6
7
8
9
10
11


### Problem 23
What will be the output of the code below?

In [23]:
for i in range(2,12):
    if i % 3 == 0:
        continue
    print i

2
4
5
7
8
10
11
